misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -164,23 +170,47 @@ private List<Path> getDependenciesForJacocoXML(JBallerinaBackend jBallerinaBacke
     /**
      * Generate the json coverage report for Testerina.
      *
-     * @param orgName package org name
-     * @param packageName package name
+     * @param orgName          package org name
+     * @param packageName      package name
      * @param filteredPathList List of the extracted source path
      * @return CoverageBuilder
      * @throws IOException
      */
     private CoverageBuilder generateTesterinaCoverageReport(String orgName, String packageName,
-                                                            List<Path> filteredPathList) throws IOException {
+                                                            List<Path> filteredPathList,
+                                                            Module originalModule) throws IOException {
         // For the Testerina report only the ballerina specific sources need to be extracted
         addCompiledSources(filteredPathList, orgName, packageName);
         execFileLoader.load(executionDataFile.toFile());
         final CoverageBuilder coverageBuilder = analyzeStructure();
-        // Create Testerina coverage report
-        createReport(coverageBuilder.getBundle(title), moduleCoverageMap);
+
+        // Analyze original modules
+        List<DocumentId> excludedFiles = getGeneratedFilesToExclude(originalModule);
+
+        // If there are files to exclude, we add that to the exclusion list
+        if (excludedFiles.isEmpty() == true) {
```

Comment:
using `==` is redundant. We can just have `if (excludedFiles.isEmpty())`

Reply:
We don't need to have an empty check here AFAIU. We can remove the condition and just pass the `excludedFiles` list